addons/mail/static/src/models/thread/thread.js

```diff
@@ -1324,6 +1340,17 @@ function factory(dependencies) {
          * interface and to notify the server of the new state.
          */
         pendingFoldState: attr(),
+        /**
+         * Reflects the process of pinning/unpinning a thread.
+         * values:
+         *     pin: the thread is pinned
+         *     pending_pin: the thread is visually pinned, and rpc haven't returned
+         *     pending_unpin: the thread is visually not pinned, but rpc haven't returned
+         *     unpin: not pinned
+         */
+        pinnedStatus: attr({
+            default: 'unpin',
+        }),
```

Comment:
Having this one variable will probably not be enough, see `foldState`. If we want the UI to update before the confirmation of the RPC, we need to store separately the pendingState and the last know serverState.
However changing the interface in such way before the confirmation of the server is dangerous and I don't like it in general (I understand that it might be better for the feeling of reactivity in the interface).

Overwriting the server state while an RPC is pending means we cannot restore the the last state that makes sense if the server refused the action (or didn't receive it).
It also means that any "new information" from the server while the RPC is pending would either be ignored or replace the pending state even though it's actually still pending.

Note: The reconciliation is actually tricky, for fold the pending state is considered done if we receive from the server the state that was requested, but in theory this might have come from another interaction. When information is coming from longpolling and not from RPC result, the request should ideally contain a unique id (isn't it already the case with json rpc) and then the server should forward that unique id when sending the answer.

Reply:
Yay github finally back. Note this assumes is_pinned is returned by the server, which will be the case in https://github.com/odoo/odoo/pull/53143

Reply:
Conceptually I understand this. There is a little subtility though with unpinning. (I checked in master before the merge, and just did it now in saas-13.3)
pinning a channel in one tab won't refresh the interface in an other

For folding, each state change is reflected in all the tabs

So, when clicking on unpin, there is a try/catch to update the interface, and when that initial rpc fails, revert to the previous state. otherwise we got what we want: the channel is not pinned and the server knows it (it won't reappear on refresh),. The bus in this case is merely a broadcaster that other tabs listen to. The opposite interaction, like I said, is not supported (maybe we should though). if another tab decides to re-pin the channel, it won't do anything in other tabs anyway

Reply:
It will crosstab/browser sync on my PR automatically as the information is always present in various channel_info sent on the bus.

Reply:
@seb-odoo 
Fair enough, then I propose to merge my PR (the way we implement it now is relevant) and adapt subsequent code in your PR to handle these use cases

Reply:
As discussed, please extract the "is_pinned" in channel_info from my PR to here.

This more or less implies:
- adding is_pinned in channel_info
- removing isPinned: true from our code, except for mail.box
- adapting tests so that channel_slots have is_pinned: true when appropriate
- override unfollow result to set is_pinned to false

If I missed anything here, diff on my PR can be used as reference.